rust/perspective-python/build.mjs

```diff
@@ -111,22 +114,96 @@ if (build_wheel) {
     cmd.sh(`maturin build ${flags} --features=${features.join(",")} ${target}`);
 }
 
-const old = fs.readFileSync("./pyproject.toml");
-
 if (build_sdist) {
-    const toml = TOML.parse(old);
-    console.log(toml);
-    delete toml.tool.maturin["data"];
-    fs.writeFileSync("./pyproject.toml", TOML.stringify(toml));
-    cmd.sh(`maturin sdist`);
+    // `maturin sdist` has some issues with Cargo workspaces, so we assemble the sdist by hand here
+    // Note that the resulting sdist is _not_ a Cargo workspace, it is rooted in this package.
+    const cargo_toml = fs.readFileSync("./Cargo.toml");
+    const pyproject_toml = fs.readFileSync("./pyproject.toml");
+    const cargo = TOML.parse(cargo_toml);
+    const pyproject = TOML.parse(pyproject_toml);
+
+    const version = cargo["package"]["version"];
+    const data_dir = `perspective_python-${version}.data`;
+    const testfile = path.join(
+        data_dir,
+        "data/share/jupyter/labextensions/@finos/perspective-jupyterlab/package.json"
+    );
+    if (!fs.existsSync(testfile)) {
+        throw new Error(
+            "labextension is not present in data directory, please build `perspective-jupyterlab`"
+        );
+    }
+    const readme_md = fs.readFileSync("./README.md");
+    const pkg_info = generatePkgInfo(pyproject, cargo, readme_md);
+    fs.writeFileSync("./PKG-INFO", pkg_info);
+    const include_paths = Array.from(cargo["package"]["include"]).concat([
```

Comment:
this is excluding some files from the `perspective` Python folder that are otherwise added by Maturin, like test arrows

https://github.com/PyO3/maturin/blob/main/src/source_distribution.rs#L624-L646

will port that

Reply:
decided against porting, instead I just expanded the set of files included with the crate in `Cargo.toml`. that way is more explicit and it's less code, to boot